tests/baselines/reference/restTupleElements1.errors.txt

```diff
@@ -18,12 +18,10 @@ tests/cases/conformance/types/tuple/restTupleElements1.ts(33,31): error TS2344:
     Type 'string' is not assignable to type 'number'.
 tests/cases/conformance/types/tuple/restTupleElements1.ts(34,31): error TS2344: Type '[number, number, string]' does not satisfy the constraint '[number, ...number[]]'.
   Type at positions 1 through 2 in source is not compatible with type at position 1 in target.
-    Type 'string | number' is not assignable to type 'number'.
-      Type 'string' is not assignable to type 'number'.
+    Type 'string' is not assignable to type 'number'.

```

Comment:
A change like this is correct with the new logic. As mentioned in the other comment - I no longer create unions from "tuple slices". That's why the reported source is just a "single" type here and not a union.

I think that perhaps this error could be somehow improved/rephrased cause it mentions a position span in the source where in fact the type mismatch happens only on a given position (or only on some positions of this span). The mentioned span covers all of the positions in the source that must match a single position of the rest in the target.

Note though that reporting a single type here is not completely new, this could always happen in cases like this:
```ts
declare let tt3: [number, string, ...string[]];
let tt4: [number, ...number[]] = tt3; 
// Type '[number, string, ...string[]]' is not assignable to type '[number, ...number[]]'.
//   Type at positions 1 through 2 in source is not compatible with type at position 1 in target.
//     Type 'string' is not assignable to type 'number'.(2322) 
```

I think there is a potential of just dropping this specific error:
https://github.com/microsoft/TypeScript/pull/50218/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R20329
in favor of the existing simpler one:
https://github.com/microsoft/TypeScript/pull/50218/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R20332

I've decided not to introduce this change here to limit the scope of the PR and to minimize the initial diff to make it easier to review it.